OvmfPkg/Library/PlatformInitLib/Platform.c

```diff
@@ -928,6 +938,43 @@ PlatformInitEmuVariableNvStore (
   Size = (UINT32)PcdGet32 (PcdFlashNvStorageVariableSize);
   ASSERT (Size < EmuVariableNvStoreSize);
 
+  //
+  // If launch a VM without OvmfFlashNvStorage device, then we'll fail

```

Comment:
Launching OVMF without variable store is not a valid configuration.

When booting with `-bios` you should use the `OVMF.fd` image (containing both CODE and VARS).